src/transformers/onnx/config.py

```diff
@@ -326,7 +401,7 @@ def num_attention_heads(self) -> int:
 
     def generate_dummy_inputs(
         self,
-        tokenizer: PreTrainedTokenizer,
+        tokenizer: "PreTrainedTokenizerBase",
```

Comment:
Are you asking about the change to `PreTrainedTokenizerBase` or use of strings for the typing? Here's the reasons in both cases:

* I chose `PreTrainedTokenizerBase` because it covers both slow and fast tokenizers. The alternative would have been something like `Union[PreTrainedTokenizer, PreTrainedTokenizerFast]`, but that felt clunky
* I used strings for the typing following @sgugger's suggestion to use the `TYPE_CHECKING` constant to fix the circular imports

Reply:
I was asking about the change of class, and it makes sense to me now, thanks for the explanation!